### ISSUE TITLE
LDSSA Partnerships - Should we consider having hackathons with companies real world data?

### ISSUE THREAD

**Issue by ivopbernardo** [LDSSA/forum#18]
Hey everyone,

I would like to discuss the possibility of having companies real data on some hackathons (similar to kaggle challenges) nearing the end of the course. I am currently working on the value proposition for our partners and I would like to have your thoughts on this.

The format would be the following:
- 1 day / weekend hackathon with a business problem and data provided by one partner. They give us a problem they are trying to solve and students will have to get their hands dirty on the raw data using everything they have learned from the course. The hackathon finishes with a business presentation with the solution. Can be an unsupervised or supervised learning problem although the unsupervised one would be harder to evaluate.

Potential pros I have identified:

1. Expose students to real world data and problems;
2. Enhance our value propositions for partners;
3. Chance to get everyone together for one day/weekend

Potential cons I have identified:
1. One day/weekend might not be enough;
2. Might interfere with all the curriculum and learning path the academy is developing;
3. Some people might not be available to be present physically.
4. It could make some people associate the academy with the company. 

Let me read your thoughts on this.

**Comment by dominguezmaria** [comment]
@ivopbernardo , really glad that we're working towards making us closer to the companies. It's something we didn't have much time to work on before, so now it feels that we're growing up!

My thought on this are the following. Let's call these hackathons, _partnership hackathons_.

If we want to open a partnership hacakthon to the students of the current batch of the Starters Academy, then such hackathon should:
* be integrated in the curriculum. We would need to evaluate if the company's data was suitable to be used in one of the hacakthons in our curriculum, and if yes, this would actually be one of the hackathons of the Starters Academy,
* only allow the participation of the students of the current batch of the Starters Academy.

This is because:
* the Starters Academy program is very intense, so they'll have their hands full already with the things we're giving them to learn,
* if we integrate a partnership hacakthon in the Starters Academy program, then we cannot have "outsiders" teaming up with the students.

Another option would be to open the partnership hackathon to anyone in the Lisbon Data Science Academy community (previous students, instructors, outsiders if they are teaming up with people from the Academy, for instance):
* it would also be good to enhance the Academy's value propositions for partners (not just the Starters Academy)
* it would be a good opportunity to bring the community together for a day /weekend
* current students of the Starters Academy would be allowed to participate, but we should clearly communicate that they should prioritise the Starters Academy over the hacakthon

To sum up: if we could integrate the partnership hacakthon in the Starters Academy curriculum, that would be ideal. Otherwise, an hackathon open to all the community also sounds like a fun option :)

**Comment by ivopbernardo** [comment]
thank you @dominguezmaria, I am looking forward to help on that chapter and create a network of partners with the same mindset and values as the Academy!

.. and thank for your comments, that was the kind of input I was looking for 👍

You referred a really great point with the possibility of giving an option to a possible partner. In the value proposition what might be interesting to approach and communicate to the companies could be something like:

**- Hackathons with partners done with one of two options:**

1. If the data and problem fits our current curriculum (TBD with all the team involved in curriculum development) we might think of an hackathon integrated in the Starters Academy;

2. Hackathon with everyone involved in the community with the objective of solving a real business problem and get everyone together.  

Our own little portuguese kaggle!

What would have to be determined to make this happen would be:
- Specific set of rules in which we decide what might fit Starters Academy curriculum (although I know that these might be more than a bunch of if-then set of rules and it might have some qualitative analysis of the problem involved).

**Comment by hershaw** [comment]
Great initiative @ivopbernardo!

@dominguezmaria's comments are spot on so I'll just add a few more specific comments:

1. When we prepare a particular dataset, we have a set of criteria for it that are very specific. It includes the number of categorical / numerical features, how big it should be, the kind of predictions, etc. so we always must find a dataset that fits the criteria so that it fits in with the course.
1. Now that I think about it, it could potentially be difficult to find someone that has a dataset that fits all of the criteria AND is something that they need solved themselves at the moment.

However, what might not be too hard is getting a dataset from a company that has used it for an interview or something else like that. Then, from the students perspective, they have a real live dataset from a real live company that they can use. It's a bit less useful for the company themselves in terms of getting actual solutions that they might be able to use themselves, though it would be an excellent thing for recruiting. They would essentially be testing ~50 people in one day on something that is relevant to their business. I understand that this would significantly change the value prop but it might be an easier sale to land.

What do you think?

**Comment by ivopbernardo** [comment]
@hershaw thank you mr! 

Well, I think that we actually should try that and pinpoint in the value prop the criteria of the possible datasets. When I start to speak with companies I should use that argument and "sell" them the opportunity of testing a recruiting methodology on the starters academy adding value to the learning process. 

Otherwise, I really like the ideia of _partnership hackathons_ that @dominguezmaria referred. It can be an excellent escape route if the company does not have a current dataset that fits the constraints. 

This second option would not be integrated in the Starters Academy and could consist of teams with current and former students and maybe even some team from the instructors side. I think that this would be a great opportunity to get everyone together, have a great time and integrate partners in the community.

Thank you for your feedback!

**Comment by hershaw** [comment]
How you feeling about this issue @ivopbernardo ? Should we keep it open? Remember that even if we close it, we can still find it and bring it back :)

**Comment by ivopbernardo** [comment]
For me, case closed @hershaw! We kinda got it solved by thinking about a final year hackathon. If any other event idea (outside the academy) comes up in the meantime we will open this again.
